File: leetcodesln/ReverseWordsInAStringIII.cs
using System.Text;

namespace leetcodesln
{
    public class ReverseWordsInAStringIII
    {
        public string ReverseWords(string s)
        {
            var sb = new StringBuilder();
            var words = s.Split(' ');
            foreach (var word in words)
            {
                sb.Append(ReverseWord(word));
                sb.Append(" ");
            }
            sb.Length--;
            return sb.ToString();
        }

        private string ReverseWord(string word)
        {
            var strarr = new char[word.Length];
            var left = 0;
            var right = word.Length - 1;
            while (left <= right)
            {
                strarr[left] = word[right];
                strarr[right] = word[left];
                left++;
                right--;
            }
            return new string(strarr);
        }
    }
}

File: Java/src/main/java/questions/ReverseWordsInAStringIII.java
package questions;

/**
 * Created by user3301 on 9/1/2017.
 *
 * Given a string, you need to reverse the order of characters in each word within a sentence while still preserving whitespace and initial word order.

 Example 1:
 Input: "Let's take LeetCode contest"
 Output: "s'teL ekat edoCteeL tsetnoc"
 */
public class ReverseWordsInAStringIII {

    public String reverseWords(String s) {
        String[] words = s.split(" ");
        StringBuffer output = new StringBuffer();
        for (String word : words) {
            output.append(new StringBuilder(word).reverse().toString());
            output.append(" ");
        }
        output.deleteCharAt(output.length()-1);
        return output.toString();
    }

    public static void main(String[] args) {
        ReverseWordsInAStringIII reverseWordsInAStringIII = new ReverseWordsInAStringIII();
        String input = "Let's take LeetCode contest";
        System.out.println(reverseWordsInAStringIII.reverseWords(input));
    }
}
